### PR TITLE
Fix sonar action can't get PR number

### PR DESCRIPTION
**Pull request by jihuayu**
We only save PR info in the CI action pull_request event.

It was tested under https://github.com/jihuayu/kvrocks-test

### CODE REVIEW

File: .github/workflows/sonar.yaml [hunk 1]
```diff
@@ -39,30 +39,45 @@ jobs:
...
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                artifact_id: matchArtifact.id,
+                archive_format: 'zip',
+              });
+              let fs = require('fs');
+              fs.writeFileSync(`${process.env.GITHUB_WORKSPACE}/${name}.zip`, Buffer.from(download.data));
+            }
+            await getData("sonarcloud-data");
+            await getData("pr");
```
Comment (git-hulk) [inline .github/workflows/sonar.yaml]: May it get the wrong PR number while two PRs are running at the same time?

Comment (jihuayu) [reply .github/workflows/sonar.yaml]: No, this is a Github action doc example.
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#using-data-from-the-triggering-workflow

File: .github/workflows/kvrocks.yaml [hunk 2]
```diff
@@ -327,7 +327,20 @@ jobs:
...
+        run: |
+          mkdir -p ./pr
+          echo "number=${{ github.event.number }}" >> ./pr/info
+          
+      - name: Upload PR info
+        uses: actions/upload-artifact@v3
+        if: ${{ matrix.sonarcloud }}
+        with:
+          name: pr
+          path: pr/
```
Comment (PragmaTwice) [inline .github/workflows/kvrocks.yaml]: You can just add a new file to `${{ env.SONARCLOUD_OUTPUT_DIR }}`.

File: .github/workflows/sonar.yaml (hunk 1 above)
Comment (git-hulk) [reply .github/workflows/sonar.yaml]: OK, thanks for your reference. That said, this artifact is only visible in the current run.

File: .github/workflows/kvrocks.yaml (hunk 2 above)
Comment (jihuayu) [reply .github/workflows/kvrocks.yaml]: You are right, Thank you. I will do it after #2049 has been merged.

File: .github/workflows/kvrocks.yaml [hunk 3]
```diff
@@ -480,4 +480,4 @@ jobs:
             exit 1
           fi
       - name: Sentinel
-        run: true
+        run: true
```
Comment (git-hulk) [inline .github/workflows/kvrocks.yaml]: Keep the empty line at the end of the file.
```suggestion
        run: true
 
```

**Review by git-hulk — COMMENTED** [review]
Looks good to me. Thanks for your great efforts.

**Review by git-hulk — APPROVED** [review]

**Pull request merged**